libs/sdk-core/src/error.rs

```diff
@@ -0,0 +1,64 @@
+use anyhow::Error;
+use std::fmt;
+
+pub type SdkResult<T, E = NewSdkError> = Result<T, E>;
+
+/// Type of error returned by the SDK
+#[derive(Debug)]
+pub enum NewSdkError {
+    SdkFailedToRequestBackup,
+
+    SdkFailedToStartBackupWatcher,
```

Comment:
In general I think we shouldn't go into too many specific errors. A structured error should let the user respond specifically and if there is not specific action that can be done then it means it is too specific.
For example I don't think we need all of `SdkNoStateSynchronizerFound`, `SdkFailedToStartBackupWatcher` `SdkInitFailedNoNodeApiOrSeedFound` as the user has not control over the flow that caused these errors. We can just use ConnectFailed and log the specific reason.

Reply:
Done